lang/java/avro/src/main/java/org/apache/avro/data/RecordBuilderBase.java

```diff
@@ -157,10 +155,7 @@ public boolean equals(Object obj) {
     if (!Arrays.equals(fieldSetFlags, other.fieldSetFlags))
       return false;
     if (schema == null) {
-      if (other.schema != null)
-        return false;
-    } else if (!schema.equals(other.schema))
-      return false;
-    return true;
+      return other.schema == null;
+    } else return schema.equals(other.schema);
```

Comment:
I like to have the braces for readability

Reply:
Done good catch thx.